cmd/otel-allocator/config/config.go

```diff
@@ -80,6 +82,101 @@ type HTTPSServerConfig struct {
 	TLSKeyFilePath  string `yaml:"tls_key_file_path,omitempty"`
 }
 
+// StringToModelDurationHookFunc returns a DecodeHookFuncType
+// that converts string to time.Duration, which can be used
+// as model.Duration.
+func StringToModelDurationHookFunc() mapstructure.DecodeHookFuncType {
```

Comment:
Is there any reason these functions need to return the actual hook function, rather than simply declaring it at such to start with? It's not like they take any parameters.

Reply:
It's fine the way it is, I was mostly curious if there was a reason worth documenting.